File: CollectionLayout/ClassGroup/ViewGroup/CollectionSectionHeaderView.m
//
//  CollectionSectionHeaderView.m
//  CollectionLayout
//
//  Created by 苏沫离 on 2020/8/11.
//  Copyright © 2020 苏沫离. All rights reserved.
//

#import "CollectionSectionHeaderView.h"

@interface CollectionSectionHeaderView ()
@property (nonatomic ,strong) UIImageView *imageView;
@end
@implementation CollectionSectionHeaderView
- (instancetype)initWithFrame:(CGRect)frame{
    self = [super initWithFrame:frame];
    if (self) {
        self.backgroundColor = UIColor.whiteColor;
        _imageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"dub_header_back"]];
        _imageView.frame = CGRectMake(0, 0, CGRectGetWidth(UIScreen.mainScreen.bounds), 172 / 375.0 * CGRectGetWidth(UIScreen.mainScreen.bounds));
        [self addSubview:_imageView];
    }
    return self;
}

@end

File: CollectionLayout-Swift/CollectionLayout-Swift/ControllerGroup/ViewGroup/CollectionSectionHeaderView.swift
//
//  CollectionSectionHeaderView.swift
//  CollectionLayout-Swift
//
//  Created by 苏沫离 on 2020/9/11.
//  Copyright © 2020 苏沫离. All rights reserved.
//

import UIKit

let kCollectionSectionHeaderIdentifer : NSString = "CollectionSectionHeaderView"

class CollectionSectionHeaderView: UICollectionReusableView {
    var imageView : UIImageView!
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = UIColor.white
        imageView = UIImageView(image: UIImage(named: "dub_header_back"))
        imageView.frame = CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height:  172 / 375.0 * UIScreen.main.bounds.width)
        self.addSubview(imageView)
    }
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}


let kCollectionSectionFooterIdentifer : NSString = "CollectionSectionFooterView"

class CollectionSectionFooterView: UICollectionReusableView {
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = UIColor.white
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
